final_task/rss_reader/rss_reader.py

```diff
@@ -0,0 +1,285 @@
+    all_links = []
+
+    soup = BeautifulSoup(article['summary'], features="html.parser")
+    for index, img in enumerate(soup.find_all('img'), 1):
+        if (str(img['src']) == '') and (str(img['alt']) == ''):
+            continue
+        if str(img['alt']) == '':
+            all_links.append(str(img['src']))
+            img.replaceWith(f"[image {index}: no description][{index}] ")
+        elif str(img['src']) == '':
+            img.replaceWith(f"[image(no link): {img['alt']}] ")
+        else:
+            all_links.append(str(img['src']))
+            img.replaceWith(f"[image {index}: {img['alt']}][{index}] ")
+
+    for link in article['links']:
+        if link['href'] not in all_links:
+            all_links.append(link['href'])
+
+    return all_links, str(soup.text)
+
+
+def unescape(text_to_unescape: str) -> str:
+    """Unescape text"""
+    return html.unescape(text_to_unescape)
+
+
+def get_articles(parsed: feedparser.FeedParserDict, limit: int) -> List[single_article.SingleArticle]:
+    """Returns list with articles."""
+    articles = []
+    feed = parsed['feed']
+    entries = parsed['entries']
+    logging.info(f'There is {len(entries)} entries')
+
+    if limit is None:
+        limit = len(entries)
+
+    try:
+        for index, entry in enumerate(entries):
+            if index == limit:
+                break
+
+            date_info = 'unknown'
+            if 'published' in entry:
+                date_info = entry['published']
+
+            links_in_article, summary_text = find_links_in_article(entry)
+
+            articles.append(
+                single_article.SingleArticle(
+                    feed=unescape(feed['title']),
+                    title=unescape(entry['title']),
+                    date=date_info,
+                    link=entry['link'],
+                    summary=unescape(summary_text),
+                    links=[f"[{num}]: {link}" for num, link in enumerate(links_in_article, 1)])
+            )
+    except KeyError as value:
+        raise custom_error.ArticleKeyError(f"One of the entries does not have {value} key")
+
+    return articles
+
+
+def print_rss_json(data: List[single_article.SingleArticle]) -> None:
+    """Prints result as JSON in stdout"""
+    print(json.dumps(data, cls=single_article.EnhancedJSONEncoder, ensure_ascii=False))
+
+
+def print_article(data: single_article.SingleArticle) -> None:
+    """Prints a single article in stdout"""
+    print("-" * 120)
+    print(data)
+    print("-" * 120)
+
+
+def print_rss_articles(rss_articles: List[single_article.SingleArticle]) -> None:
+    """Prints source feed and articles"""
+    for article in rss_articles:
+        print_article(article)
+
+
+def check_the_connection(rss_url: str) -> str:
+    """Checks the connection"""
+    info = ''
+    try:
+        urllib.request.urlopen(rss_url)
+    except urllib.request.HTTPError as e:
+        info = f'{e.code}: {e.reason}'
+        return info
```

Comment:
A question

Do you really need to return an empty string here in cace of exception?